### PR TITLE
Code clean up of ClientWebSocketChannel files

### PR DESCRIPTION
**Pull request by drwill-ms**
More incremental changes leading up the the CA1001 fix, to track down what is causing test run failures in the pipeline.

These classes look awfully similar but are used in 2 different projects, and one is public (why? I don't know).

### CODE REVIEW

File: common/src/device/transport/mqtt/ClientWebSocketChannel.cs [hunk 1]
```diff
@@ -13,24 +14,20 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport.Mqtt
 {
-#pragma warning disable CA1001 // Types that own disposable fields should be disposable - WS is owned by the caller.
-
+    //[SuppressMessage("Design", "CA1001:Types that own disposable fields should be disposable", Justification = "WS is owned by the caller.")]
```
Comment (drwill-ms) [inline common/src/device/transport/mqtt/ClientWebSocketChannel.cs:17]: I have this ready, but will disable it until actually dealing with `CancellationTokenSource`

File: common/src/device/transport/mqtt/ClientWebSocketChannel.cs [hunk 2]
```diff
@@ -13,24 +14,20 @@
...
     internal class ClientWebSocketChannel : AbstractChannel
-#pragma warning restore CA1001 // Types that own disposable fields should be disposable
     {
-        private readonly ClientWebSocket _webSocket;
-        private readonly CancellationTokenSource _writeCancellationTokenSource;
-        private bool _active;
-
-        internal bool ReadPending { get; set; }
```
Comment (drwill-ms) [inline common/src/device/transport/mqtt/ClientWebSocketChannel.cs:25]: These weren't referenced outside this class

File: common/src/device/transport/mqtt/ClientWebSocketChannel.cs [hunk 3]
```diff
@@ -84,7 +75,67 @@ protected override void Flush0()
             }
         }
 
-        protected override bool IsCompatible(IEventLoop eventLoop) => true;
+        public ClientWebSocketChannel Option<T>(ChannelOption<T> option, T value)
```
Comment (drwill-ms) [inline common/src/device/transport/mqtt/ClientWebSocketChannel.cs:78]: Moved up from below for publics above protected/internal/private

**Review by azabbasi — APPROVED** [review]

**Pull request merged**